thrall/app/controllers/ReaperController.scala

```diff
@@ -143,23 +144,24 @@ class ReaperController(
     logger.info(s"Hard deleting next $count images...")
 
     (for {
-      idsHardDeletedFromES: Set[String] <- es.hardDeleteNextBatchOfImages(isReapable, count)
-      mainImagesS3Deletions <- store.deleteOriginals(idsHardDeletedFromES)
-      thumbsS3Deletions <- store.deleteThumbnails(idsHardDeletedFromES)
-      pngsS3Deletions <- store.deletePNGs(idsHardDeletedFromES)
-      idsNotProcessedInDynamo <- softDeletedMetadataTable.clearStatuses(idsHardDeletedFromES)
+      BatchDeletionIds(esIds, esIdsActuallyDeleted) <- es.hardDeleteNextBatchOfImages(isReapable, count)
+      mainImagesS3Deletions <- store.deleteOriginals(esIdsActuallyDeleted)
+      thumbsS3Deletions <- store.deleteThumbnails(esIdsActuallyDeleted)
+      pngsS3Deletions <- store.deletePNGs(esIdsActuallyDeleted)
+      idsNotProcessedInDynamo <- softDeletedMetadataTable.clearStatuses(esIdsActuallyDeleted)
     } yield {
-      if (idsHardDeletedFromES.isEmpty) {
+      if (esIds.isEmpty) {
         logger.info(s"Although $count images were requested to be hard deleted, none were found to be hard deletable.")
       }
-      metrics.hardReaped.increment(n = idsHardDeletedFromES.size).run
-      idsHardDeletedFromES.map { id =>
+      metrics.hardReaped.increment(n = esIdsActuallyDeleted.size).run
+      esIds.map { id =>
+        val wasHardDeletedFromES = esIdsActuallyDeleted.contains(id)
         val detail = Map(
-          "ES" -> Some(true), // since this is list of IDs deleted from ES
+          "ES" -> Some(wasHardDeletedFromES),
           "mainImage" -> mainImagesS3Deletions.get(ImageIngestOperations.fileKeyFromId(id)),
           "thumb" -> thumbsS3Deletions.get(ImageIngestOperations.fileKeyFromId(id)),
           "optimisedPng" -> pngsS3Deletions.get(ImageIngestOperations.optimisedPngKeyFromId(id)),
-          "dynamo" -> Some(!idsNotProcessedInDynamo.contains(id)),
+          "dynamo.table.softDelete.metadata" -> (if(wasHardDeletedFromES) Some(!idsNotProcessedInDynamo.contains(id)) else None)
```

Comment:
oh just to make it clearer that all we're deleting is the record from the soft delete metadata table, rather than deleting all traces of the image from Dynamo (which we don't do anymore since https://github.com/guardian/grid/pull/4130, which in part is to facilitate [restoring from replica](https://github.com/guardian/grid/pull/4128)).

We do need to delete the soft deleted records though otherwise that table will get silly big over time

Reply:
Sorry, completely missed that this is just for logging / json response